### PR TITLE
Add output limits to pid manager

### PR DESCRIPTION
**Pull request by aschokking**
defaults to -1 1

### CODE REVIEW

**Comment by aschokking** [comment]
@JohnGilb

File: src/xbot/common/math/PIDManager.java [hunk 1]
```diff
@@ -13,26 +13,37 @@
...
-            double defaultD) {
-        propP = new DoubleProperty(functionName + " P", defaultP, propMan);
-        propI = new DoubleProperty(functionName + " I", defaultI, propMan);
-        propD = new DoubleProperty(functionName + " D", defaultD, propMan);
+            double defaultD, double defaultMaxOutput, double defaultMinOutput) {
+        propP = propMan.createPersistentProperty(functionName + " P", defaultP);
+        propI = propMan.createPersistentProperty(functionName + " I", defaultI);
+        propD = propMan.createPersistentProperty(functionName + " D", defaultD);
+        
+        maxOutput = propMan.createPersistentProperty(functionName + " MaxOutput", defaultMaxOutput);
```
Comment (WasabiFan) [inline src/xbot/common/math/PIDManager.java]: There should be spaces between words

File: src/xbot/common/math/PIDManager.java [hunk 2]
```diff
@@ -13,26 +13,37 @@
...
-        setupPIDManager(functionName, propMan, defaultP, defaultI, defaultD);
+        setupPIDManager(functionName, propMan, defaultP, defaultI, defaultD, 1.0, -1.0);
     }
 
     public PIDManager(String functionName, XPropertyManager propMan) {
-        setupPIDManager(functionName, propMan, 0, 0, 0);
+        setupPIDManager(functionName, propMan, 0, 0, 0, 1.0, -1.0);
     }
 
     private void setupPIDManager(String functionName, XPropertyManager propMan, double defaultP, double defaultI,
```
Comment (WasabiFan) [inline src/xbot/common/math/PIDManager.java]: Why do we have this method? I think this should go in the most verbose constructor and then the others should just cascade down from least to most explicit.

**Comment by JohnGilb** [comment]
lgtm

File: src/xbot/common/math/PIDManager.java (hunk 1 above)
Comment (aschokking) [inline src/xbot/common/math/PIDManager.java]: I'm not sure we ever really settled on a philosophy for what things should look like in the property names. I feel like most of the robot ones are camel cased for instance.

File: src/xbot/common/math/PIDManager.java (hunk 2 above)
Comment (aschokking) [inline src/xbot/common/math/PIDManager.java]: Sounds reasonable, not sure why it was done this way originally

File: src/xbot/common/math/PIDManager.java (hunk 1 above)
Comment (WasabiFan) [inline src/xbot/common/math/PIDManager.java]: There isn't any technical reason to omit spaces (and you include them to separate the value from the label already). So, given that humans are better-able to read space-separated words, we should target the most readable option.

**Pull request merged**